pkgs/development/python-modules/qiskit/default.nix

```diff
@@ -1,76 +1,115 @@
 {
+  stdenv,
   lib,
   pythonOlder,
   buildPythonPackage,
   fetchFromGitHub,
+  cargo,
+  rustPlatform,
+  rustc,
+  libiconv,
 
-  # build-system
+  dill,
+  numpy,
+  python-dateutil,
+  rustworkx,
+  scipy,
   setuptools,
+  setuptools-rust,
+  stevedore,
+  symengine,
+  sympy,
+  typing-extensions,
 
-  # Python Inputs
-  qiskit-aer,
-  qiskit-ibmq-provider,
-  qiskit-ignis,
-  qiskit-terra,
-  # Optional inputs
-  withOptionalPackages ? true,
-  qiskit-finance,
-  qiskit-machine-learning,
-  qiskit-nature,
-  qiskit-optimization,
-  # Check Inputs
-  pytestCheckHook,
+  withVisualization ? false,
+  matplotlib,
+  pillow,
+  pydot,
+  pylatexenc,
+  seaborn,
+
+  withCrosstalkPass ? false,
+  z3-solver,
+
+  withCspLayoutPass ? false,
+  python-constraint,
 }:
 
 let
-  optionalQiskitPackages = [
-    qiskit-finance
-    qiskit-machine-learning
-    qiskit-nature
-    qiskit-optimization
+  visualizationPackages = [
+    matplotlib
+    pillow
+    pydot
+    pylatexenc
+    seaborn
   ];
+  crosstalkPackages = [ z3-solver ];
+  cspLayoutPackages = [ python-constraint ];
```

Comment:
Are those optional features? If so we should place them under optional-dependencies.